server/jetstream_cluster.go

```diff
@@ -7999,9 +8077,27 @@ func (mset *stream) stateSnapshot() []byte {
 // Grab a snapshot of a stream for clustered mode.
 // Lock should be held.
 func (mset *stream) stateSnapshotLocked() []byte {
+	var consumers []*consumerAssignment
+	if mset.cfg.ManagesConsumers {
+		consumers = make([]*consumerAssignment, 0, len(mset.consumers))
+		for _, o := range mset.consumers {
+			// Skip if the consumer is pending, we can't include it in our snapshot.
+			// If the proposal fails after we marked it pending, it would result in a ghost consumer.
+			if o.ca == nil || o.ca.pending {
```

Comment:
We're reading `consumerAssignment` here, should we hold the JS read lock?
We're also reading from the consumer, should we hold the consumer read lock?

Reply:
Probably, but need to figure out how not to violate the locking order here...

Reply:
Believe we don't need to be holding it here. Since the `ManagesConsumers` setting can't be updated, and we only grab the data here when doing a snapshot as part of `applyStreamEntries`, we should be good. Any changes to the assignment will only be made inside `applyStreamEntries` anyway.